cirq/ops/gate_features.py

```diff
@@ -45,9 +45,15 @@ def on_each(self, *targets: raw_types.Qid) -> op_tree.OP_TREE:
             Operations applying this gate to the target qubits.
 
         Raises:
-            ValueError if targets are not instances of Qid.
+            ValueError if targets are not instances of Qid or List[Qid].
         """
-        return [self.on(target) for target in targets]
+        operations = []
+        for target in targets:
+            if isinstance(target, list):
```

Comment:
We shouldn't be specializing to list. If we're going to flatten, we should flatten everything.

If it's a qubit id, apply the operation.
It it's a collections.Iterable, recurse.
If it's neither, or if it's a known problematic type like string (which is an infinitely descending iterable), raise an error.

Add a test making sure the string case terminates. Add a test that provides qubits using a generator.

Reply:
I followed your suggestions in the newest push.